src/ArmTemplates/Extractor/EntityExtractors/GatewayApiExtractor.cs

```diff
@@ -67,10 +72,14 @@ public class GatewayApiExtractor : IGatewayApiExtractor
 
                 if (!string.IsNullOrEmpty(singleApiName))
                 {
-                    // inluding only apis with singleApiName
-                    var apis = gatewayApis.Where(x => x.Name == singleApiName);
+                    var serviceApi = await this.apiClientUtils.GetsingleApi(singleApiName, extractorParameters);
+
+                    // inluding only api with singleApiName
+                    var apis = gatewayApis.Where(x => x.Name == serviceApi.Name).ToList();
```

Comment:
Should we keep casing/culture in mind here and use equals?

Reply:
Since the results are retrieved from the same api and processed by the same data processor after retrieving data, it should be equal. 
Thank you for raising that.